pandas/io/tests/test_parsers.py

```diff
@@ -4128,6 +4130,34 @@ def test_bool_header_arg(self):
             with tm.assertRaises(TypeError):
                 pd.read_fwf(StringIO(data), header=arg)
 
+    def test_multithread_stringio_read_csv(self):
+        max_row_range = 10000
+        num_files = 100
+
+        bytes = ['\n'.join(['%d,%d,%d' % (i, i, i) for i in range(max_row_range)]).encode()
+                 for j in range(num_files)]
+        files = [BytesIO(b) for b in bytes]
+
+        # Read all files in many threads
+        pool = ThreadPool(8)
+        results = pool.map(pd.read_csv, files)
+        first_result = results[0]
+
+        for result in results:
+            tm.assert_frame_equal(first_result, result)
+
+    def test_multithread_path_read_csv(self):
+        df = DataFrame(np.random.rand(5000, 20), columns=map(str, xrange(20)))
+        num_files = 20
+        with tm.ensure_clean('__passing_str_as_dtype__.csv') as path:
+            df.to_csv(path, index=False)
+            pool = ThreadPool(4)
+            read_dataframes = pool.map(pd.read_csv, [path]*num_files)
+
+            for single_dataframe in read_dataframes:
+                tm.assert_frame_equal(df, single_dataframe, check_names=False)
+
+
 class TestMiscellaneous(tm.TestCase):
```

Comment:
note AFAICT this DOES work!. I just want to have this as a test for validation.

Like to test this with actual files and ``StringIO``